include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -359,6 +361,9 @@ __parallel_transform_reduce(_ExecutionPolicy&& __exec, _Up __u, _LRp __brick_lea
                     if (__local_idx == 0)
                     {
                         __temp_acc[__offset_1 + __item_id.get_group(0)] = __result;
+                        //store result in one-element buffer
+                        if (__item_id.get_group(0) == 0)
```

Comment:
I would also check for `__n_groups==1` so that the code is easier to understand

Reply:
yes, perhaps  it reasonable. I'll look into it.